lib/wasi/src/lib.rs

```diff
@@ -22,8 +22,8 @@ mod utils;
 use crate::syscalls::*;
 
 pub use crate::state::{
-    Fd, WasiFile, WasiFs, WasiFsError, WasiState, WasiStateBuilder, WasiStateCreationError,
-    ALL_RIGHTS, VIRTUAL_ROOT_FD,
+    Fd, HostFile, Pipe, Stderr, Stdin, Stdout, WasiFile, WasiFs, WasiFsError, WasiState,
```

Comment:
I think it's probably fine to make `Stdout`, `Stderr`, and `Stdin` public but I'm not sure about `HostFile`.

Is it useful to make `HostFile` public? My thoughts are that it's very likely that it's a type full of weird implementation details that we probably don't want to have to not break

Reply:
I can imagine some niche uses for it, like piping the output of some WASM to a file on the host machine, but I don't mind keeping that private for now. If we add the `Pipe`, then users should still be able to pipe a file themselves.